project/src/test/java/com/google/sps/data/DataAccessorTest.java

```diff
@@ -162,6 +261,34 @@ public void storeUserPreferences_emptydUserId_throwIllegalArgumentException() {
         () -> dataAccessor.storeUserPreferences("", userPrefs));
   }
 
+  private Entity createRecomEntityByProperties(String userId, String placeId, boolean chosen,
+  boolean tryAgain) {
+    Entity recommendationEntity = new Entity(DataAccessor.RECOMMENDATION_ENTITY_KIND);
+    recommendationEntity.setProperty(DataAccessor.USER_ID_PROPERTY, userId);
+    recommendationEntity.setProperty(DataAccessor.PLACE_ID_PROPERTY, placeId);
+    recommendationEntity.setProperty(DataAccessor.TIME_PROPERTY, System.currentTimeMillis());
+    recommendationEntity.setProperty(DataAccessor.TRY_AGAIN_PROPERTY, tryAgain);
+    recommendationEntity.setProperty(DataAccessor.CHOSEN_PROPERTY, chosen);
+    return recommendationEntity;
+  }
+
+  private List<Entity> getRecommendationEntitiesByUserId(String userId) {
+    return datastoreService.prepare(
+        new Query(DataAccessor.RECOMMENDATION_ENTITY_KIND)
+            .setFilter(new Query.FilterPredicate("UserId", FilterOperator.EQUAL, userId))
+    ).asList(FetchOptions.Builder.withDefaults());
+  }
+
+  private UserFeedback buildUserFeedback(String userId, String chosenPlace,
+      ImmutableList<String> places, boolean userTriedAgain) {
+    UserFeedback.Builder temp = UserFeedback
+        .builder()
+        .setUserId(userId)
+        .setRecommendedPlaces(places)
+        .setUserTriedAgain(userTriedAgain);
+    return chosenPlace == null ? temp.build() : temp.setChosenPlace(chosenPlace).build();
```

Comment:
For readability sake, I think you should separate this to two statements: 

```
if (chosenPlace != null) { temp.setChosenPlace(...); }
return chosenPlace.build();
```

Reply:
Great tip. done